packages/mui-system/src/colorManipulator/colorManipulator.js

```diff
@@ -1,62 +1,41 @@
 /* eslint-disable @typescript-eslint/naming-convention */
-import clamp from '@mui/utils/clamp';
 import MuiError from '@mui/internal-babel-macros/MuiError.macro';
+import clampProduction from '@mui/utils/clamp';
+import * as Color from '../color';
+import { getRed, getGreen, getBlue, getAlpha } from '../color';
 
-/**
- * Returns a number whose value is limited to the given range.
- * @param {number} value The value to be clamped
- * @param {number} min The lower boundary of the output range
- * @param {number} max The upper boundary of the output range
- * @returns {number} A number in the range [min, max]
- */
-function clampWrapper(value, min = 0, max = 1) {
-  if (process.env.NODE_ENV !== 'production') {
-    if (value < min || value > max) {
-      console.error(`MUI: The value provided ${value} is out of range [${min}, ${max}].`);
-    }
-  }
+const clamp = process.env.NODE_ENV === 'production' ? clampProduction : clampDevelopment;
 
-  return clamp(value, min, max);
+function clampDevelopment(value, min = 0, max = 1) {
+  if (value < min || value > max) {
+    console.error(`MUI: The value provided ${value} is out of range [${min}, ${max}].`);
+  }
+  return clampProduction(value, min, max);
 }
 
 /**
  * Converts a color from CSS hex format to CSS rgb format.
  * @param {string} color - Hex color, i.e. #nnn or #nnnnnn
- * @returns {string} A CSS rgb color string
+ * @returns {string} A CSS rgb() or rgba() color string
  */
-export function hexToRgb(color) {
-  color = color.slice(1);
-
-  const re = new RegExp(`.{1,${color.length >= 6 ? 2 : 1}}`, 'g');
-  let colors = color.match(re);
-
-  if (colors && colors[0].length === 1) {
-    colors = colors.map((n) => n + n);
-  }
-
-  return colors
-    ? `rgb${colors.length === 4 ? 'a' : ''}(${colors
-        .map((n, index) => {
-          return index < 3 ? parseInt(n, 16) : Math.round((parseInt(n, 16) / 255) * 1000) / 1000;
-        })
-        .join(', ')})`
-    : '';
-}
-
-function intToHex(int) {
-  const hex = int.toString(16);
-  return hex.length === 1 ? `0${hex}` : hex;
+export function hexToRgb(hexadecimal) {
+  return Color.formatRGB(Color.parse(hexadecimal));
 }
 
 /**
  * Returns an object with the type and values of a color.
- *
- * Note: Does not support rgb % values.
- * @param {string} color - CSS color, i.e. one of: #nnn, #nnnnnn, rgb(), rgba(), hsl(), hsla(), color()
+ * Supported formats:
+ * - #hhh, #hhhhhh, #hhhhhhhh
+ * - rgb(), rgba()
+ * - hsl(), hsla()
+ * - color() with one of: 'srgb', 'display-p3', 'a98-rgb', 'prophoto-rgb', 'rec-2020'
+ * @param {string} color - CSS color in one of the supported formats
  * @returns {object} - A MUI color object: {type: string, values: number[]}
+ * @deprecated
```

Comment:
Is the reason is to use the new `Color.parse` instead of this?

Reply:
So my goal is to guide users away from slow manipulation functions at runtime, and this one is slow.

I see that it's used internally in the theme code, and that's ok as far as I can tell because theme setup happens once, so there's no issue with keeping it internally (because we know when to use it).

I'm not sure if there is a good use case for this for end-users unlike other utility functions like `darken` or `alpha`, that's why I marked it.

Reply:
Thanks for the info.